### PR TITLE
refactor: update fee spec with pre and post conditions

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

File: interbtc-spec/docs/source/spec/fee.rst [hunk 1]
```diff
@@ -95,89 +78,79 @@ Fee for users to premium redeem (as percentage). If users execute a redeem with
...
 
 Specification
 .............
 
 *Function Signature*
 
-``distributeRewards()``
+``distributeRewards(amount)``
 
+*Preconditions*
```
Comment (nud3l) [inline interbtc-spec/docs/source/spec/fee.rst:118]: Looks like pre-conditions are missing here. For this to pass, I assume we have to have at least 1 vault registered and active?

File: interbtc-spec/docs/source/spec/fee.rst [hunk 2]
```diff
@@ -95,89 +78,79 @@ Fee for users to premium redeem (as percentage). If users execute a redeem with
...
-Function Sequence
-.................
-
-1. Calculate the total fees for all Vaults using the `VaultRewards` percentage.
-2. Calculate the total fees for all Staked Relayers using the `StakedRelayerRewards` percentage.
-3. Calculate the total fees for all Collators using the `CollatorRewards` percentage.
-4. Send the remaining fees to the Maintainer fund.
+*Postconditions*
 
+* The Vault reward pool MUST increase by ``amount * VaultRewards``.
```
Comment (nud3l) [inline interbtc-spec/docs/source/spec/fee.rst:122]: Where is the Vault reward pool defined?

File: interbtc-spec/docs/source/spec/fee.rst [hunk 3]
```diff
@@ -95,89 +78,79 @@ Fee for users to premium redeem (as percentage). If users execute a redeem with
...
-A function that allows Staked Relayers, Vaults and Collators to withdraw the fees earned.
+A function that allows incentivised network participants to withdraw the fees earned.
 
 Specification
 .............
 
 *Function Signature*
 
-``withdrawRewards(account, currency, amount)``
+``withdrawRewards(account)``
```
Comment (nud3l) [inline interbtc-spec/docs/source/spec/fee.rst:137]: They can only withdraw all or nothing? If so, please add to the function description above.

File: interbtc-spec/docs/source/spec/fee.rst [hunk 4]
```diff
@@ -95,89 +78,79 @@ Fee for users to premium redeem (as percentage). If users execute a redeem with
...
-.................
+*Preconditions*
+
+* The ``account`` MUST have available rewards for ``interBTC``.
+
+*Postconditions*
 
-1. Compute the rewards based on the account's stake.
-2. Transfer all rewards to the account.
+* The account's balance MUST increase by the available rewards.
```
Comment (nud3l) [inline interbtc-spec/docs/source/spec/fee.rst:153]:
```suggestion
* The account's balance MUST increase by the available rewards.
* The account's withdrawable rewards MUST decrease by the withdrawn rewards.
```

File: interbtc-spec/docs/source/spec/fee.rst (hunk 2 above)
Comment (gregdhill) [reply interbtc-spec/docs/source/spec/fee.rst:122]: Will need to link to the `reward` component when that is defined.

File: interbtc-spec/docs/source/spec/fee.rst [hunk 5]
```diff
@@ -95,89 +78,82 @@ Fee for users to premium redeem (as percentage). If users execute a redeem with
...
-.................
+* There MUST be at least one registered Vault.
 
-1. Calculate the total fees for all Vaults using the `VaultRewards` percentage.
-2. Calculate the total fees for all Staked Relayers using the `StakedRelayerRewards` percentage.
-3. Calculate the total fees for all Collators using the `CollatorRewards` percentage.
-4. Send the remaining fees to the Maintainer fund.
+*Postconditions*
 
+* The Vault reward pool MUST increase by ``amount * VaultRewards``.
```
Comment (nud3l) [inline interbtc-spec/docs/source/spec/fee.rst:124]:
```suggestion
 .. todo:: link to reward pool
 
* The Vault reward pool MUST increase by ``amount * VaultRewards``.
```

**Pull request merged**